### ISSUE TITLE
Getting weird error "Assertion failed: Shader input  is not present." which doesn't really help figuring out what input is missing

### ISSUE THREAD

**Issue by JonasT** [panda3d/panda3d#157]
I am getting the weird error "Assertion failed: Shader input  is not present." which doesn't really help figuring out what input is missing. i managed to nail it down to the following code to reproduce (requires "test.egg" model of any sort present which can be loaded into the scene so the render actually renders the shader):

```python
#!/usr/bin/python3

from panda3d.core import KeyboardButton, Shader,\
    Texture, Vec3, PythonCallbackObject, loadPrcFileData,\
    PointLight
from direct.showbase.ShowBase import ShowBase

loadPrcFileData('', "gl-version 4 0")

base = ShowBase()

shader = Shader.make(Shader.SLGLSL,
"""
#version 330

void main() {
}
""", """
#version 330

struct p3d_LightSourceParameters {
  vec4 position;
  vec4 diffuse;
  vec4 specular;
  vec3 attenuation;
  vec3 spotDirection;
  float spotCosCutoff;
  float spotExponent;
  sampler2DShadow shadowMap;
  mat4 shadowMatrix;
};
uniform p3d_LightSourceParameters shadowLight0;

out vec4 p3d_FragColor;

void main() {
  p3d_FragColor.rgb += shadowLight0.position.xyz;
}
""")

base.render.set_shader(shader)

_l = PointLight("Shadowless point light")
l = base.render.attach_new_node(_l)
l.node().set_scene(base.render)
l.node().set_active(True)
base.render.set_light(l)
base.render.setShaderInput('shadowLight0', l)

m = base.loader.loadModel("test.egg")
m.reparentTo(base.render)

base.run()
```
I suspect something about the struct members might be wrong, but without a better error message it is really hard to tell what the issue is..

TL;DR: would it be possible to provide a better error message here? That would be neat!

**Comment by rdb** [comment]
You are using the deprecated `shadowMatrix` input, and the code to emulate its functionality has a bug.  It's easy to fix, and I'll check it in soon.

However, you should really be using the `shadowViewMatrix`, which you can use to transform the view-space vertex position instead of the model-space position.  This is more efficient since Panda then has to do far fewer matrix multiplications per frame.
